source/Body.cpp

```diff
@@ -33,21 +33,22 @@ using namespace std;
 
 
 // Constructor, based on a Sprite.
-Body::Body(const Sprite *sprite, Point position, Point velocity, Angle facing, double zoom)
-	: position(position), velocity(velocity), angle(facing), zoom(zoom), sprite(sprite), randomize(true)
+Body::Body(const Sprite *sprite, Point position, Point velocity, Angle facing, double zoom, Point scale)
+	: position(position), velocity(velocity), angle(facing), scale(scale), zoom(zoom), sprite(sprite), randomize(true)
 {
 }
 
 
 
 // Constructor, based on the animation from another Body object.
-Body::Body(const Body &sprite, Point position, Point velocity, Angle facing, double zoom)
+Body::Body(const Body &sprite, Point position, Point velocity, Angle facing, double zoom, Point scale)
 {
 	*this = sprite;
 	this->position = position;
 	this->velocity = velocity;
 	this->angle = facing;
 	this->zoom = zoom;
+	this->scale = scale;
```

Comment:
// Be sure to keep the existing scale for this Body sprite and scale that by this new local scale.
	this->scale = scale * sprite.Scale();

Reply:
See also: https://github.com/xobes/endless-sky/tree/thrust-bwoof for updates with endless-sky/master